client/packages/reports/src/components/ReportWidget.tsx

```diff
@@ -23,66 +32,110 @@ export const ReportWidget: React.FC<PropsWithChildren<ReportWidgetProps>> = ({
   title,
   Icon,
   reports,
-}) => (
-  <Card
-    sx={{
-      borderRadius: '16px',
-      height,
-      maxWidth,
-      padding: '24px',
-      display: 'flex',
-      flexDirection: 'column',
-      margin: '11px',
-      flex: 1,
-      boxShadow: theme => theme.shadows[2],
-    }}
-  >
-    <Grid container alignItems="center">
-      <Grid
-        display="flex"
-        alignItems="center"
-        justifyContent="center"
-        sx={{ width: 48 }}
-      >
-        <Icon color="primary" />
+  onReportClick,
+  reportWithArgs,
+  setReportWithArgs,
+}) => {
+  const navigate = useNavigate();
+
+  const { setId, setArgs } = useReportStore();
+
+  const reportArgs = useCallback(
+    (report: ReportRowFragment, args: JsonData | undefined) => {
+      setArgs(report.id, args);
+      navigate(
+        RouteBuilder.create(AppRoute.Reports).addPart(report.id).build()
+      );
+    },
+    [setArgs, navigate]
+  );
+
+  return (
+    <Card
+      sx={{
+        borderRadius: '16px',
+        height,
+        maxWidth,
+        padding: '24px',
+        display: 'flex',
+        flexDirection: 'column',
+        margin: '11px',
```

Comment:
:nit because were are in an `sx` prop, we can use numbers, which are multipliers of the default border etc - based on screen size, so they can be responsive:
```suggestion
        borderRadius: 4,
        height,
        maxWidth,
        padding: 3,
        display: 'flex',
        flexDirection: 'column',
        margin: 1.375,
```